sensors/depth/depth.go

```diff
@@ -0,0 +1,42 @@
+// Package depth implements the Depth sensor
+package depth
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/registry"
+
+	"go.viam.com/slam/sensors/utils"
+)
+
+// Depth represents a depth sensor.
+type Depth struct {
+	Name  string
+	depth camera.Camera
+}
+
+// New creates a new Depth sensor based on the sensor definition and the service config.
+func New(ctx context.Context, deps registry.Dependencies, sensors []string, sensorIndex int) (Depth, error) {
```

Comment:
[note to self] We expect specific sensors to be at specific indexes in the config (e.g. the config in app.viam.com's config tab). We might want to replace this array with a dictionary.